message/nimessage.yml

```diff
@@ -0,0 +1,415 @@
+      responses:
+        101:
+          description: Switching Protocols
+          headers:
+            Sec-WebSocket-Accept:
+              description: Used in the WebSocket opening handshake. Set to a value derived from the Sec-WebSocket-Key header.
+              type: string
+        426:
+          description: Upgrade Required
+
+definitions:
+  Error:
+    description: Contains error information
+    type: object
+    properties:
+      name:
+        description: The string error code
+        type: string
+      code:
+        description: Numeric error code
+        type: integer
+      resourceType:
+        description: The type of resource associated with the error
+        type: string
+      resourceId:
+        description: The identifier of the resource associated with the error
+        type: string
+      message:
+        description: The filled in error message
+        type: string
+      args:
+        description: Positional argument values for the error code
+        type: array
+        items:
+          type: string
+      innerErrors:
+        type: array
+        items:
+          $ref: '#/definitions/Error'
+    example:
+      name: Skyline.OneOrMoreErrorsOccurred
+      code: -251040
+      message: One or more errors occurred. See the contained list for details of each error.
+      args: []
+      innerErrors:
+        - name: Message.SessionTokenNotFound
+          code: -251608
+          resourceType:
+          resourceId:
+          message: Session token 'f755d531-2403-433f-a0b7-463ccb8f2b9' not found.
+          args: ['f755d531-2403-433f-a0b7-463ccb8f2b9']
+  TopicAndMessage:
+    type: object
+    title: Topic and Message
+    required:
+      - message
+      - topic
+    properties:
+      topic:
+        description: The topic name
+        type: string
+      message:
+        description: The message content
+        type: string
+    description: The topic and message content
+  Topic:
+    type: object
+    title: Topic
+    required:
+      - topic
+    properties:
+      topic:
+        description: The topic name
+        type: string
+    description: The topic name
+  SessionToken:
+    type: object
+    title: Session Token
+    required:
+      - token
+    properties:
+      token:
+        description: A unique session ID
+        type: string
+    description: A unique session ID
+  QueueSizeBytes:
+    type: object
+    title: Current Queue Size
+    required:
+      - queueSizeBytes
+    properties:
+      queueSizeBytes:
+        description: The current queue size (in bytes)
+        type: integer
+    description: The current queue size (in bytes)
+  MaxQueueSizeBytes:
+    type: object
+    title: Maximum Queue Size
+    required:
+      - maxQueueSizeBytes
+    properties:
+      maxQueueSizeBytes:
+        description: The maximum queue size (in bytes)
+        type: integer
+    description: The maximum queue size (in bytes)
+  WebSocketPublish:
```

Comment:
This seems odd to have 3 different "Action" objects that define single-element ENUMs.  It may be better to have a single "WebSocket Action" object, and then have an "action" ENUM that has ["publish", "subscribe", "unsubscribe"], and then have the description explain how the different actions behave.  That would mirror the actual implementation.

Reply:
Dropping per our offline conversation